File: business/business-oj/src/main/java/com/glowxq/oj/judge/enums/SubmitType.java
package com.glowxq.oj.judge.enums;

import com.glowxq.core.common.enums.base.BaseType;
import com.glowxq.oj.submit.compile.impl.CompileInteractiveSubmitImpl;
import com.glowxq.oj.submit.compile.impl.CompileSpjSubmitImpl;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author glowxq
 * @version 1.0
 * @date 2025/1/20
 */
@AllArgsConstructor
@Getter
public enum SubmitType implements BaseType {

    /**
     * 普通评测
     */
    NormalJudge("NormalJudge", "普通评测", null),

    /**
     * 客观题评测
     */
    FixedJudge("FixedJudge", "普通评测", null),

    /**
     * 远程评测
     */
    RemoteJudge("RemoteJudge", "远程评测", null),

    /**
     * 在线调试
     */
    TestJudge("TestJudge", "在线调试", null),

    /**
     * 编译特判程序
     */
    CompileSpj("CompileSpj", "编译特判程序", CompileSpjSubmitImpl.class),

    /**
     * 编译交互程序
     */
    CompileInteractive("CompileInteractive", "编译交互程序", CompileInteractiveSubmitImpl.class);

    /**
     * code
     */
    private final String code;

    /**
     * 最近类型
     */
    private final String name;

    private final Class<?> clazz;

    /**
     * 根据code获取枚举
     *
     * @param code
     * @return
     */
    public static SubmitType matchCode(String code) {
        for (SubmitType sceneType : SubmitType.values()) {
            if (sceneType.getCode().equals(code)) {
                return sceneType;
            }
        }
        return null;
    }
}

File: oj-web/src/enums/oj/problem/SubmitType.ts
import type { BaseEnum } from '@/enums/base';

// 提交评测类型枚举接口
interface SubmitJudgeTypeEnum extends BaseEnum {
  tooltip?: string;
  text?: string;
}

/**
 * 提交评测类型枚举
 * 用于标识不同的评测方式
 */
export const SubmitType: { [key: string]: SubmitJudgeTypeEnum } = {
  /**
   * 普通评测
   */
  NORMAL_JUDGE: {
    code: 'NormalJudge',
    name: '普通评测',
    tooltip: '普通评测',
    text: '标准评测流程'
  },

  /**
   * 客观题评测
   */
  FIXED_JUDGE: {
    code: 'FixedJudge',
    name: '客观题评测',
    tooltip: '客观题评测',
    text: '针对选择、填空等客观题的评测'
  },

  /**
   * 远程评测
   */
  REMOTE_JUDGE: {
    code: 'RemoteJudge',
    name: '远程评测',
    tooltip: '远程评测',
    text: '使用远程评测系统进行评测'
  },

  /**
   * 在线调试
   */
  TEST_JUDGE: {
    code: 'TestJudge',
    name: '在线调试',
    tooltip: '在线调试',
    text: '用于测试和调试代码'
  },

  /**
   * 编译特判程序
   */
  COMPILE_SPJ: {
    code: 'CompileSpj',
    name: '编译特判程序',
    tooltip: '编译特判程序',
    text: '编译用于特殊判题的程序'
  },

  /**
   * 编译交互程序
   */
  COMPILE_INTERACTIVE: {
    code: 'CompileInteractive',
    name: '编译交互程序',
    tooltip: '编译交互程序',
    text: '编译用于交互式题目的程序'
  }
};

/**
 * 根据code获取枚举
 * @param code
 * @returns 枚举值或null
 */
export function matchCode(code: string): SubmitJudgeTypeEnum | null {
  return Object.values(SubmitType).find(item => item.code === code) || null;
}
